### DISCUSSION TITLE
Replace/Add Domain Function In My Filters

### DISCUSSION THREAD

**Discussion by BlazeFTL** [uBlockOrigin/uBlock-issues#3212] [Ideas] [answered]
Currently Site Owners Buy Domains In Bulk(Mostly Shortners) Because Its Cheap

Would It Be Possible To Add Replace Domains In My Filters

Ex : I Type Google.com In A Box And In Another Box I Would Type The Replacement Domain (google2.com)
```
Google.com##ads ==> google2.com##ads
```

Another Function - Add Domain
So If I Type Google.com  In A Box And In Another Box I Would Type google2.com
Now With Add Button
So The Results Would Be 
```
Google.com##ads ==> google.com,google2.com##ads
+
@@*$ghide,domain=google.com ==> @@*$ghide,domain=google.com|google2.com
```
https://github.com/AdguardTeam/Scriptlets/issues/427

**Comment by BlazeFTL** [comment]
@gorhill Sir Is It Possible ?

**Comment by ghost** [comment, marked as answer]
Looks similar to this https://github.com/uBlockOrigin/uBlock-issues/issues/330

**Reply by BlazeFTL** [reply]
Yes That Idea Is Also Good
Just Copy Pasting Domains Is So Hectic Now Because Every Site Owner Is Doing This For Evading AdBlock And It Would Also Reduce Filter Maintainers Burden

!#define Gplinks (example.com,example2.com)
Gplinks##.ads
!#endif

So Just Pasting Domain Once Is Enough😄

**Comment by uBlock-user** [comment]
Use regex - `/google[0-9]{1}\.com/##.ads`

**Reply by BlazeFTL** [reply]
Unfortunately Site Owners Dont Buy Domains Like That
Ex https://github.com/uBlockOrigin/uAssets/commit/7a9edb06431ce974be75a4ff0f43879c649eecd5
Regex Wont Work For Any Of These Cases
For Example Gplinks Have Evaded Adblockers By Changing Domains 1-2 Times Every Day(Its Always New Domains)(Like One Time Use)
This Method Seems To Be The Best Method To Evade Adblockers(Ofc Domain Buying Is Causing Them A Little Money Losses)

**Reply by uBlock-user** [reply]
> Ex https://github.com/uBlockOrigin/uAssets/commit/7a9edb06431ce974be75a4ff0f43879c649eecd5

based on that, they make use of `.com`, `.in` domains and occasionally `.blog`

`com,in,blog##.ads`

**Reply by garry-ut99** [reply]
There is a reason why `##.ads` doesn't exist in EasyList as a generic filter- either it was causing too many breakages or/and was triggering too many anti-adblocks.

**Reply by BlazeFTL** [reply]
You Misunderstood
##.ads It Was Just A Example For The Idea Of https://github.com/uBlockOrigin/uBlock-issues/issues/330

**Reply by uBlock-user** [reply]
> There is a reason why ##.ads doesn't exist in EasyList as a generic filter- either it was causing too many breakages or/and was triggering too many anti-adblocks.

What makes you think I'm focusing on that ?

**Reply by garry-ut99** [reply]
I overlooked the context.

**Comment by krystian3w** [comment]
Maybe the solution is some Android text editor with find and replace function?

Then you wouldn't need to update code mirror in uBo to implement input with replace function like on GitHub:

![Screenshot_20240511-140430](https://github.com/uBlockOrigin/uBlock-issues/assets/35370833/13f55988-2d42-4ae3-a816-b19af96a5e81)

**Reply by BlazeFTL** [reply]
I am aware about that
You Simply Dont Have Any Idea About How Many Domains Have To Be Updated Now Everyday
 Also you just keep spamming dislike everywhere ?

**Reply by BlazeFTL** [reply]
For Example I Use This Many Lines For 1 Site
So I Have To Update Domain If There Is A New Domain
Just Think About The Time It Consumes To Copy Paste(Cant Keep Export/Importing For 10+ Sites With Text Editor)
Although It Was Meant To Reduce Burden On Filter Maintainers Not For Personal Usage https://github.com/uBlockOrigin/uBlock-issues/issues/330
```
dropgalaxy.*,financemonk.net##+js(rpnt, script, /.js'|.js"|e.cloudfront.net/, false)
dropgalaxy.*,financemonk.net###downloadBtnClick:remove-attr(disabled)
@@||publisher.ad-maven.com/assets/theme/vendor/modernizr/modernizr.js$script,domain=dropgalaxy.*|financemonk.net
@@||dash.ezmob.com/ss/js/ads.js$script,domain=dropgalaxy.*|financemonk.net
dropgalaxy.*,financemonk.net##+js(set, isAdblockActive, )
||tmp.isavetube.com/js/countd.js$script,3p,domain=dropgalaxy.*|financemonk.net,replace=/throw\s+new\s+Error\(\);?//g
||tmp.isavetube.com/js/countd.js$script,3p,domain=dropgalaxy.*|financemonk.net,replace=/window\.getComputedStyle\([^)]+\)/window.getComputedStyle(false)/g
||dropgalaxy.com^$replace=/(\\$\(".*?"\)\.val\()'1'(\);)/\$1'0'\$2/,~script,~xmlhttprequest,~stylesheet
!dropgalaxy.*,financemonk.net##+js(acs, document.documentElement, break;case $.)
!dropgalaxy.*,financemonk.net##+js(rmnt, script, ;}}};break;case $.)
!dropgalaxy.*,financemonk.net##+js(rmnt, script, /h=decodeURIComponent|popundersPerIP/)
dropgalaxy.*,financemonk.net##+js(nostif, checkLostFocus)
dropgalaxy.*,financemonk.net##+js(set, document.hidden, false)
dropgalaxy.*,financemonk.net###method_free:style(display: block !important;)
dropgalaxy.*,financemonk.net##+js(set, badip, false)
dropgalaxy.*,financemonk.net###apkad, #adm-container-4010, #method_free2:style(clip-path: circle(0) !important;)
||tmp.isavetube.com/js/countd.js$script,3p,domain=dropgalaxy.*|financemonk.net,replace=/\.height>\d+&&/.height>=0&&/
||tmp.isavetube.com/js/countd.js$script,3p,domain=dropgalaxy.*|financemonk.net,replace=/\.length>\d+&&/.length>=0&&/
!||tmp.isavetube.com/js/countd.js$script,3p,domain=dropgalaxy.*|financemonk.net,replace=/<= 0/=> 0/
!||tmp.isavetube.com/js/countd.js$script,3p,domain=dropgalaxy.*|financemonk.net,replace=/.*?getComputedStyle[\s\S]*?/!0 ?/gms
!||tmp.isavetube.com/js/countd.js$script,3p,domain=dropgalaxy.*|financemonk.net,replace=/.*?getPropertyValue[\s\S]*?/!0 ?/gms
!||tmp.isavetube.com/js/countd.js$script,3p,domain=dropgalaxy.*|financemonk.net,replace=/window\.getComputedStyle\(.+\)\.getPropertyValue/!0 ?/gms
!dropgalaxy.*,financemonk.net#@#+js()
||youtube.com^$domain=dropgalaxy.*|financemonk.net,important
dropgalaxy.*,financemonk.net##.mt-5.col-md-12, .footer, .btn-block.btn-danger.btn, .text-primary, li, #article:style(clip-path: circle(0) !important;)
dropgalaxy.*,financemonk.net##.row, #download0_alt, #downloadBtnClick:style(margin-top: -90px !important;)
dropgalaxy.*,financemonk.net##.downloadNow, #dl:style(margin-top: -550px !important;)
dropgalaxy.*,financemonk.net##.btn-sm.btn-primary.btn-block.btn.downloadbtn, .legalWrap.box, .subtitle, .title, #faq:style(clip-path: circle(0) !important;)
dropgalaxy.*,financemonk.net##.message.page:style(margin-top: 50px !important;)
dropgalaxy.*,financemonk.net##+js(trusted-click-element, #hideSlideBanner)
dropgalaxy.*,financemonk.net###dllink, #dl:style(display: block !important;)
dropgalaxy.*,financemonk.net##+js(nano-sib, downloadTimer, *)
/images/dgpromo.jpg$image,domain=dropgalaxy.*|financemonk.net
/tum-dono-ki-mkc-meme.gif$image,domain=dropgalaxy.*|financemonk.net
||fastly.jsdelivr.net/npm/disable-devtool/disable-devtool.min.js$script,domain=dropgalaxy.*|financemonk.net
!dropgalaxy.*,financemonk.net##button, #countdown:style(background: linear-gradient(to right,#eb3349 40%,#f45c43) ! important;)
!@@*$ehide,domain=dropgalaxy.*|financemonk.net
```
